privacyidea/lib/utils/__init__.py

```diff
@@ -1123,27 +1125,30 @@ def check_pin_policy(pin, policy):
     """
     chars = {"c": r"[a-zA-Z]",
              "n": r"[0-9]",
-             "s": r"[\[\].:,;_<>+*!/()=?$§%&#~^-]"}
+             "s": SPECIAL_CHARS_REGXEP}
     ret = True
     comment = []
 
     if not policy:
         return False, "No policy given."
+    if not pin:
+        return False, "No pin given."
 
-    if policy[0] in ["+", "-"] or policy[0] is not "[":
+    # this is necessary since a policy "cns" equals "+cns"
```

Comment:
This is not true.

cns means it needs to contain a character, a number and a special character.

+cns means it needs to contain something from the group of characters, numbers  and special characters. 

Why was it necessary to touch the check_pin_policy in the first place.

Reply:
Maybe not. I misread the documentation in the sense of grouping (the imnportant example comes after the line break at https://privacyidea.readthedocs.io/en/latest/policies/admin.html#otp-pin-contents). I'll check.

Reply:
https://github.com/privacyidea/privacyidea/pull/2187/commits/40740a7eb1610400b44c7e0b811ee9944101ab67 should resolve the errors.